File: C_BasicTrader/src/CSVReader.cpp
#include "CSVReader.h"

#include "helper/UsefullFunctions.h"
#include "helper/ConfigManager.h"

#include <vector>
#include <string>
#include <stdlib.h>
#include <iostream>
#include <sstream>
#include "PriceFeedData.h"

using namespace std;

int CSVReader::readExchangeFromFile(PriceFeedData &prices, string filepath)
{
    string csvFile = filepath;
    string line;
    string thisVal;
    char cvsSplitBy = ',';
    ifstream inputFile;
    functions::openInputFile(inputFile, config::configValues["exchangeInputDir"], filepath);

    std::getline(inputFile, line, '\n');

    while (std::getline(inputFile, line, '\n'))
    {
        stringstream lineStream(line);
        vector<string> splitLine;
        while (getline(lineStream, thisVal, cvsSplitBy))
        {
            splitLine.emplace_back(thisVal);
        }
        prices.addPrice(atof(splitLine[1].c_str()) + 0.0001, atof(splitLine[1].c_str()) - 0.0001, atof(splitLine[0].c_str()) * 1000);
    }
    inputFile.close();
    return true;
}

File: Java/BasicTrader/CSVReader.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.LinkedList;

public class CSVReader{

    public static PriceFeedData readExchangeFromFile(String filepath)
    {
        String csvFile = filepath;
        String line = "";
        String cvsSplitBy = ";";
        PriceFeedData prices = new PriceFeedData();

        try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
            
            br.readLine();

            while ((line = br.readLine()) != null) {

                // use comma as separator
                String[] splitLine = line.split(cvsSplitBy);

                prices.addPrice(Float.valueOf(splitLine[2])+0.0001, Float.valueOf(splitLine[2])-0.0001, Long.valueOf(splitLine[1])*1000);
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
        return prices;
    }
}